File: sequentialDigits.py
# https://leetcode.com/problems/sequential-digits/ 

class Solution(object):
	def sequentialDigits(self, low, high):
		res = []
		while low <= high:
			num = str(low)
			newlow = str(int(num[0]) + 1)
			for _ in range(1, len(num)):
				newlow += "0"

			digs = num[0]
			for _ in range(1, len(num)):
				d = int(digs[-1]) + 1
				if d > 9:
					digs = ""
					break
				else:
					digs += str(d)

			if digs:
				digs = int(digs)
				if low <= digs <= high:
					res.append(digs)

			low = int(newlow)

		return res

File: sequentialDigits.java
// https://leetcode.com/problems/sequential-digits/ 

/*
 * You can find my solution to this problem at sequentialDigits.py.
 * After I solve a problem, I like to examine other people's solutions.
 * I found a very nice solution here:
 * https://leetcode.com/problems/sequential-digits/discuss/853592/Python-Solution-using-queue-explained
 * I wanted to preserve that solution, and I wanted to rewrite it in Java:
 */

class Solution {
	public List<Integer> sequentialDigits(int low, int high) {
		Queue<Integer> queue = new LinkedList<Integer>();
		for (int num = 1; num <= 9; num++) {
			queue.add(num);
		}

		List<Integer> res = new ArrayList<Integer>();
		while (!queue.isEmpty()) {
			int num = queue.poll();
			if (num >= low && num <= high) {
				res.add(num);
			}

			int last = num % 10;
			if (last < 9) {
				queue.add(num * 10 + last + 1);
			}
		}

		return res;
	}
}
